master/internal/api_workspace.go

```diff
@@ -69,17 +69,17 @@ func validateWorkspaceName(name string) error {
 func (a *apiServer) GetWorkspaceByID(
 	ctx context.Context, id int32, curUser model.User, rejectImmutable bool,
 ) (*workspacev1.Workspace, error) {
-	notFoundErr := status.Errorf(codes.NotFound, "workspace (%d) not found", id)
 	w := &workspacev1.Workspace{}
 
 	if err := a.m.db.QueryProto("get_workspace", w, id, curUser.ID); errors.Is(err, db.ErrNotFound) {
-		return nil, notFoundErr
+		return nil, status.Errorf(codes.NotFound, "workspace (%d) not found", id)
 	} else if err != nil {
 		return nil, errors.Wrapf(err, "error fetching workspace (%d) from database", id)
 	}
 
 	if err := workspace.AuthZProvider.Get().CanGetWorkspace(ctx, curUser, w); err != nil {
-		return nil, authz.SubIfUnauthorized(err, notFoundErr)
+		return nil, authz.SubIfUnauthorized(err,
+			status.Errorf(codes.NotFound, "workspace (%d) not found, please check your permissions.", id))
```

Comment:
I think this gives it away that the workspace id exists.

Reply:
The language for our errors is: 
```
project (%d) not found
workspace (%d) not found
trial (%d) not found
task (%d) not found
```
and so on. For the RBAC-permission errors I just add `please check your permissions`. If we wanted to change this language across the board, perhaps try `Missing project/workspace/trial/task (%d). If you think this is an error, please check your permissions.`

Reply:
sorry, I should have added more context. I mean could this in conjunction with line 75 give it away that the workspace exists but you don't have permission to know it does?